apps/src/lib/jsdebugger/JsDebugger.jsx

```diff
@@ -143,10 +152,60 @@ var UnconnectedJsDebugger = React.createClass({
 
   getInitialState() {
     return {
-      watchersHidden: false
+      watchersHidden: false,
+      open: true,
     };
   },
 
+  componentDidMount() {
+    this.props.debuggerUi.initializeAfterDomCreated({
+      defaultStepSpeed: this.props.stepSpeed,
+      root: this.root,
+      component: this,
+    });
+  },
+
+  isOpen() {
+    return this.state.open;
+  },
+
+  slideShut() {
+    const closedHeight = $(this.root).find('#debug-area-header').height() + $(this.root).find('#debugResizeBar').height();
+    this.setState({
+      transitionType: 'closing',
+      open: false,
+      openedHeight: $(this.root).height(),
+      closedHeight,
+    });
+    $('#codeTextbox').animate(
```

Comment:
Yeah I thought about that. But at the same time, if you don't know about the lifecycle of the dom node, you could end up referring to a dom node that no longer exists. Perhaps it would be better to invert the control here and pass in a `onResize` prop to `JsDebugger` which did this animation bit?